include/Reader/readerir.h

```diff
@@ -1317,6 +1325,7 @@ class GenIR : public ReaderBase {
   uint32_t TargetPointerSizeInBits;
   const uint32_t UnmanagedAddressSpace = 0;
   const uint32_t ManagedAddressSpace = 1;
+  static uint32_t TailCallCount;
 };
```

Comment:
CoreCLR always assumes that the JIT is multithreading safe. It would be up to LLILC to add a giant lock around the compile method if this assumption does not hold.